File: 0169-majority-element/0169-majority-element.c
int majorityElement(int* nums, int numsSize) {
    int temp;
    int count=0;
    for(int i =0;i<numsSize;i++)
    {
        if(count==0)
        {
            count=1;
            temp=nums[i];
        }
        else if(nums[i]==temp)
        {
            count++;
        }
        else count--;
    }
    return temp;
}

File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        HashMap<Integer,Integer> map = new HashMap<>();
        int n = nums.length;
        for(int i=0;i<n;i++){
            if(map.containsKey(nums[i])){
                map.put(nums[i],map.get(nums[i])+1);
            }
            else{
                map.put(nums[i],1);
            }
        }

        for(Map.Entry<Integer,Integer> entry : map.entrySet()){
            if(entry.getValue() > n/2){
                return entry.getKey();
            }
        }
        return -1;
    }
}